broker/broker.go

```diff
@@ -66,20 +66,19 @@ type ChainInfoI interface {
 	GetChainID() (string, error)
 }
 
-type OrderEvent interface {
-	events.Event
-	Order() *types.Order
-}
-
 type OrderEventWithVegaTime struct {
-	OrderEvent
+	events.Order
 	vegaTime time.Time
 }
```

Comment:
Why not just change your OrderEvent interface instead of directly embedding events.Order? (which I believe is the go way to make testing easier etc...)

Reply:
OK I don't quite understand but I believe you :)